### ISSUE TITLE
Offering help if needed

### ISSUE THREAD

**Issue by X-Ryl669** [gnumpi/esphome_audio#27]
Hi,

 I've been linked to your repository via [here](https://github.com/esphome/firmware/pull/173#issuecomment-2051222537). 
I've implemented an update to ESP-ADF component to support the latest version and sharing I2C bus with ESP-ADF's format and ESPHome. My work is [here](https://github.com/X-Ryl669/esphome/tree/esp-adf-take2

I think it can be useful to merge my changes (at least for using the last ESP-ADF and dependencies version). There were a lot of pitfalls I had to fix (like the esp-dsp dependency, the I2C issue, the esp-adf incompatibility with C and C++ structure definitions and so on). So it might be save you a lot of time to avoid redoing this.

Let me know if you need it.

**Comment by gnumpi** [comment]
Hey, sounds really interesting. Do I get it right that supporting the ADF-I2C let you use the DACs and ADCs supported by the ADF-boards? I also added some DAC and ADC support to the I2S_audio component in this repository. Would be awesome to add the ADF-boards right away. Could you share a config which makes use of your change to get a better understanding?

**Comment by X-Ryl669** [comment]
> Do I get it right that supporting the ADF-I2C let you use the DACs and ADCs supported by the ADF-boards?

Yes. The (HW) components supported by ADF are now accessible directly and no need to replicate those in ESPHome, while still maintaining access to ESPHome's I2C devices (like AHT sensor on BOX3, Radar, etc...).

> Could you share a config which makes use of your change to get a better understanding?

https://github.com/esphome/issues/issues/5296#issuecomment-2039156195

For the change itself, please have a look [here](https://github.com/esphome/esphome/pull/6486), it's not too big in fact
And BTW, Espressif merged my change to ADF to support external I2C bus integration, so the upcoming versions will have it without a link to my fork.

**Comment by gnumpi** [comment]
Really great news! I will try to integrate the [esp_codec_dev](https://github.com/espressif/esp-adf/tree/master/components/esp_codec_dev) devices into the `adf_pipeline` component then. 
Thanks for offering your help! If it's ok, I will come back to you when I have a rough beta version and see what you think.
